server/runtime/src/main/java/org/infinispan/server/configuration/security/KeyStoreConfiguration.java

```diff
@@ -40,9 +41,10 @@
 public class KeyStoreConfiguration extends ConfigurationElement<KeyStoreConfiguration> {
    static final AttributeDefinition<String> ALIAS = AttributeDefinition.builder(Attribute.ALIAS, null, String.class).build();
    static final AttributeDefinition<String> GENERATE_SELF_SIGNED_CERTIFICATE_HOST = AttributeDefinition.builder(Attribute.GENERATE_SELF_SIGNED_CERTIFICATE_HOST, null, String.class).build();
-   static final AttributeDefinition<Supplier<char[]>> KEY_PASSWORD = AttributeDefinition.builder(Attribute.KEY_PASSWORD, null, (Class<Supplier<char[]>>) (Class<?>) Supplier.class)
-         .serializer(AttributeSerializer.SECRET).build();
-   static final AttributeDefinition<Supplier<char[]>> KEYSTORE_PASSWORD = AttributeDefinition.builder(Attribute.PASSWORD, null, (Class<Supplier<char[]>>) (Class<?>) Supplier.class)
+   @Deprecated
+   static final AttributeDefinition<Supplier<CredentialSource>> KEY_PASSWORD = AttributeDefinition.builder(Attribute.KEY_PASSWORD, null, (Class<Supplier<CredentialSource>>) (Class<?>) Supplier.class)
```

Comment:
well, if it is deprecated, can't we just remove the code?

Reply:
I've just deprecated it :)